typescript/cli/src/deploy/utils.ts

```diff
@@ -96,3 +100,54 @@ export function isISMConfig(
 export function isZODISMConfig(filepath: string): boolean {
   return parseIsmConfig(filepath).success;
 }
+
+export async function prepareDeploy(
+  multiProvider: MultiProvider,
+  userAddress: Address,
+  chains: ChainName[],
+  dryRun: boolean = false,
+): Promise<Record<string, BigNumber>> {
+  const initialBalances: Record<string, BigNumber> = {};
+  await Promise.all(
+    chains.map(async (chain: ChainName) => {
+      const provider = dryRun
+        ? getLocalProvider()
+        : multiProvider.getProvider(chain);
+      const currentBalance = await provider.getBalance(userAddress);
+      initialBalances[chain] = currentBalance;
+    }),
+  );
+  return initialBalances;
+}
+
+export async function completeDeploy(
+  command: Command,
+  initialBalances: Record<string, BigNumber>,
+  multiProvider: MultiProvider,
+  userAddress: Address,
+  chains: ChainName[],
+  dryRun: boolean = false,
+) {
+  if (chains.length > 0) logPink(`⛽️ Gas Usage Statistics`);
+  for (const chain of chains) {
+    const provider = dryRun
+      ? getLocalProvider()
+      : multiProvider.getProvider(chain);
+    const currentBalance = await provider.getBalance(userAddress);
+    const balanceDelta = initialBalances[chain].sub(currentBalance);
+    if (dryRun && balanceDelta.lt(0)) break;
+    logPink(
+      `\t- Gas used during ${command} ${
+        dryRun ? 'dry-run' : 'deploy'
+      } on ${chain}: ${balanceDelta} wei or ${ethers.utils.formatEther(
+        balanceDelta,
+      )} ether`,
```

Comment:
nit
```suggestion
      `\t- Balance used during ${command} ${
        dryRun ? 'dry-run' : 'deploy'
      } on ${chain}: ${ethers.utils.formatEther(
        balanceDelta,
      )} ether`,
```
idk if we need both formats
also would be nice to use the `chainMetadata.nativeToken.symbol`/`decimals` instead of `ether`

Reply:
We're using `ethers` to calculate balance and all EVM-compatible chains have a native decimals of 18 and are denominated in wei, so this doesn't seem pertinent

Re symbol I do agree, so will update this